client/pages/me/me.js

```diff
@@ -1,66 +1,98 @@
 // pages/me/me.js
-Page({
+const qcloud = require('wafer2-client-sdk');
+const app = getApp();
 
-  /**
-   * 页面的初始数据
-   */
+Page({
   data: {
-
+    personal: {},
+    chineseName: null,
+    chineseNameInputValue: null,
+    englishName: null,
+    englishNameInputValue: null,
+    phone: null,
+    phoneInputValue: null,
+    email: null,
+    emailInputValue: null,
+    btnDisable: true
   },
-
-  /**
-   * 生命周期函数--监听页面加载
-   */
-  onLoad: function (options) {
-
+  onLoad: function () {
+    this.setData({
+      personal: app.userInfo
+    });
   },
 
-  /**
-   * 生命周期函数--监听页面初次渲染完成
-   */
-  onReady: function () {
-
+  chineseNameInput: function (event) {
+    if (event.detail.value.length > 0) {
+      this.btnDisable = false
+      this.setData({
+        chineseNameInputValue: event.detail.value,
+        btnDisable: false
+      });
+    }
   },
-
-  /**
-   * 生命周期函数--监听页面显示
-   */
-  onShow: function () {
-
+  englishNameInput: function (event) {
+    if (event.detail.value.length > 0) {
+      this.btnDisable = false
+      this.setData({
+        englishNameInputValue: event.detail.value,
+        btnDisable: false
+      });
+    }
   },
-
-  /**
-   * 生命周期函数--监听页面隐藏
-   */
-  onHide: function () {
-
+  phoneInput: function (event) {
+    if (event.detail.value.length > 0) {
+      this.btnDisable = false
+      this.setData({
+        phoneInputValue: event.detail.value,
+        btnDisable: false
+      });
+    }
   },
-
-  /**
-   * 生命周期函数--监听页面卸载
-   */
-  onUnload: function () {
-
-  },
-
-  /**
-   * 页面相关事件处理函数--监听用户下拉动作
-   */
-  onPullDownRefresh: function () {
-
+  emailInput: function (event) {
+    if (event.detail.value.length > 0) {
+      this.btnDisable = false
+      this.setData({
+        emailInputValue: event.detail.value,
+        btnDisable: false
+      });
+    }
   },
 
-  /**
-   * 页面上拉触底事件的处理函数
-   */
-  onReachBottom: function () {
-
-  },
-
-  /**
-   * 用户点击右上角分享
-   */
-  onShareAppMessage: function () {
-
+  infoSaveClick: function (event) {
+    this.setData({
+      chineseName: this.data.chineseNameInputValue,
+      englishName: this.data.englishNameInputValue,
+      phone: this.data.phoneInputValue,
+      email: this.data.emailInputValue,
+      btnDisable: true
+    });
+    //update the latest info into server
+    qcloud.request({
+      url: 'http://localhost:5757',//config.service.userUrl + '?OpenID=' + that.userInfo.openId,
+      method: 'POST',
+      header: {
+        'content-type':'application/x-www-form-urlencoded'
+      },
+      data: {
+        //OpenID: this.personal.openId (using annotation to avoid error of undefined openID)
```

Comment:
I think the OpenId field is needed.

Reply:
I think the method should be "PUT" instead.

Reply:
But isn't "PUT" used for only updating and not creating new items?

Reply:
What we're doing at this endpoint is : create if not exists and update if it exists. It's an idempotent method. Using put should be fine.
https://stackoverflow.com/questions/630453/put-vs-post-in-rest
https://en.wikipedia.org/wiki/Idempotence

Reply:
Got it!